adlchecker/src/main/java/com/nedap/archie/adlchecker/TerminologyContentGenerator.java

```diff
@@ -0,0 +1,218 @@
+                    addCodeToTerminology(adlContent, resultingArchetype, atCode);
+                }
+            }
+        }
+    }
+
+    private void sortTerminology(Archetype resultingArchetype) {
+        Map<String, Map<String, ArchetypeTerm>> termDefinitions = resultingArchetype.getTerminology().getTermDefinitions();
+        for(String language: termDefinitions.keySet()) {
+            Map<String, ArchetypeTerm> stringArchetypeTermMap = termDefinitions.get(language);
+            Map<String, ArchetypeTerm> sortedArchetypeTermMap = new TreeMap<>(new Comparator<String>() {
+                @Override
+                public int compare(String o1, String o2) {
+                    String code1 = o1.substring(0, 2);
+                    String code2 = o2.substring(0, 2);
+                    if(code1.equalsIgnoreCase(code2)) {
+                        int number1 = Integer.parseInt(o1.substring(2));
+                        int number2 = Integer.parseInt(o2.substring(2));
+                        return number1 - number2;
+                    } else {
+                        return code2.compareTo(code1);
+                    }
+                }
+            });
+            sortedArchetypeTermMap.putAll(stringArchetypeTermMap);
+            termDefinitions.put(language, sortedArchetypeTermMap);
+
+        }
+    }
+
+    private void walkArchetype(String sourceFile, Archetype archetype, Archetype resultingArchetype) {
+        LinkedList<CObject> workList = new LinkedList<>();
+        workList.push(archetype.getDefinition());
+        List<CObject> cObjects = new ArrayList<>();
+        while(!workList.isEmpty()) {
+            CObject next = workList.pop();
+            if(!terminologyHasCodeForAllLanguages(archetype, next)) {
+                cObjects.add(next);
+                addCodeToTerminology(sourceFile, resultingArchetype, next.getNodeId());
+            }
+            for(CAttribute attribute:next.getAttributes()) {
+                for(CObject child:attribute.getChildren()) {
+                    workList.push(child);
+                    if(child instanceof CTerminologyCode) {
+                        checkTerminologyCode(sourceFile, resultingArchetype, (CTerminologyCode) child);
+                    }
+                }
+            }
+        }
+    }
+
+    private void checkTerminologyCode(String sourceFile, Archetype archetype, CTerminologyCode child) {
+        for(String constraint:child.getConstraint()) {
+            if(!terminologyHasCodeForAllLanguages(archetype, constraint)) {
+                addCodeToTerminology(sourceFile, archetype, constraint);
+            }
+        }
+    }
+
+    private void addCodeToTerminology(String sourceFile, Archetype resultingArchetype, String code) {
+        String text = getCommentName(sourceFile, code);
+        if(text == null) {
+            text = "Add term for me!";
+        }
+        String description = text; //TODO: grep comment from sourcefile!
```

Comment:
Isn't the todo done by getCommentName()?

Reply:
fixed in next branch